sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -46,7 +47,8 @@ class UnresolvedException[TreeType <: TreeNode[_]](tree: TreeType, function: Str
 case class UnresolvedRelation(
     multipartIdentifier: Seq[String],
     options: CaseInsensitiveStringMap = CaseInsensitiveStringMap.empty(),
-    override val isStreaming: Boolean = false)
+    override val isStreaming: Boolean = false,
+    partitions: Seq[Partition] = Seq.empty[Partition])
```

Comment:
Is `partitions` only for the JDBC V2 path? If so, could we use `options` instead for passing partition info into the JDBC scan?

Reply:
Yes, `partitions` is only for JDBC V2 path, but it seems to me that we can't use `options` to pass the partition info into JDBC Scan. In `JDBCTable.newScanBuilder`, it merges `jdbcOptions` with `options` in `CaseInsensitiveStringMap`, but partitions is not a `JDBCOptions`, so it won't be passed to `JDBCScanBuilder`. I am not sure if we should add partitions into `JDBCOptions`.

Reply:
IMHO adding it in `JDBCOptions` looks better than adding it in `UnresolvedRelation` because the partition info is dedicated for JDBC scans and having physical partition info in a class field of the logical plan is a bit weird. But, its okay to wait for the comments of other reviewers.